tests/Doctrine/Tests/ORM/Functional/Ticket/DDC6470Test.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Tools\ToolsException;
+
+/**
+ */
+class DDC6470Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        $this->enableSecondLevelCache();
+        parent::setUp();
+
+        try {
+            $this->_schemaTool->createSchema(
+                [
+                    $this->_em->getClassMetadata(DDC6470User::class),
+                    $this->_em->getClassMetadata(DDC6470Driver::class),
+                ]
+            );
+        } catch (ToolsException $exc) {
+        }
+    }
+
+    public function testIssue()
+    {
+        $driver1 = new DDC6470Driver('Bar');
+
+        $this->_em->persist($driver1);
+        $this->_em->flush();
+        $this->_em->clear();
+
+
+        $this->assertTrue($this->_em->getCache()->containsEntity(DDC6470User::class, ['id' => $driver1->getUserProfile()->getId()]));
+        $queryCount = $this->getCurrentQueryCount();
+        $driver3 = $this->_em->createQueryBuilder()
```

Comment:
@shustrik the problem with this test is that your query is not cacheable, therefore it won't load data from L2C.

Your test pass if you call `QueryBuilder#setCacheable(true)` or if you just use `$driver3 = $this->_em->find(DDC6470Driver::class, $driver1->getId());` here.